src/lucky_migrator/create_table_statement.cr

```diff
@@ -58,24 +58,33 @@ class LuckyMigrator::CreateTableStatement
 
   # Generates raw sql from a type declaration and options passed in as named
   # variables.
-  macro add(type_declaration, index = false, using = :btree, unique = false)
+  macro add(type_declaration, index = false, using = :btree, unique = false, **type_options)
+    {% options = type_options.empty? ? nil : type_options %}
+
     {% if type_declaration.type.is_a?(Union) %}
-      add_column :{{ type_declaration.var }}, {{ type_declaration.type.types.first }}, optional: true
+      add_column :{{ type_declaration.var }}, {{ type_declaration.type.types.first }}, optional: true, options: {{ options }}
     {% else %}
-      add_column :{{ type_declaration.var }}, {{ type_declaration.type }}
+      add_column :{{ type_declaration.var }}, {{ type_declaration.type }}, options: {{ options }}
     {% end %}
 
     {% if index || unique %}
       add_index :{{ type_declaration.var }}, using: {{ using }}, unique: {{ unique }}
     {% end %}
   end
 
-  def add_column(name, type : (String | Time | Int32 | Int64 | Float | Bool).class, optional = false, reference = nil, on_delete = :do_nothing)
+  def add_column(name, type : (String | Time | Int32 | Int64 | Float | Bool).class, optional = false, reference = nil, on_delete = :do_nothing, options : NamedTuple? = nil)
+
+    if options.is_a?(NamedTuple)
```

Comment:
I think doing `if options` to check for `nil` is more common. I don't feel strongly though so if you prefer this, that is also fine!

Reply:
I didn't know about `if options` so I tried `options.nil?` which didn't work properly. But this is cleaner, its more like JS